### PR TITLE
[AKS] fix aks pr check-in test pipeline

### PR DESCRIPTION
**Pull request by FumingZhang**
Base on [PR#3415](https://github.com/Azure/azure-cli-extensions/pull/3415), this PR further fix some other known issues of the AKS PR check-in test pipeline. The main changes are as follows:

* Update az_aks_tool, add path check to avoid unexpected errors caused by invalid path
* Fix the problem that the variable passed to az_aks_tool in the extensions test script is lost
* Add two new test cases excluded by default (test_aks_disable_local_accounts, test_aks_update_to_msi_cluster_with_addons)
* Add a new pipeline variable *EXT_TEST_MATRIX* to override the default filtering rules

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/aks-preview/az_aks_tool/main.py [hunk 1]
```diff
@@ -67,6 +67,9 @@ def main():
     print("raw args: {}".format(sys.argv))
```
[the remark this answers is not in the document]
Comment (FumingZhang) [reply src/aks-preview/az_aks_tool/main.py:67]: Update key changes in PR [description section](https://github.com/Azure/azure-cli-extensions/pull/3389#issue-645740121).

File: src/aks-preview/azcli_aks_live_test/setup_venv.sh [hunk 2]
```diff
@@ -1,18 +1,27 @@
...
 # delete existing venv
 rm -rf azEnv || true
 
 # install python packages
-python$PYTHON_VERSION -m venv azEnv
+python${PYTHON_VERSION} -m venv azEnv
 source azEnv/bin/activate
 python -m pip install -U pip
-# fixed azdev version to avoid call failure in az_aks_tool
-pip install azdev==0.1.32
```
Comment (andyliuliming) [inline src/aks-preview/azcli_aks_live_test/setup_venv.sh:15]: please keep sticking to one version.

File: src/aks-preview/azcli_aks_live_test/ext_matrix_default.json [hunk 3]
```diff
@@ -21,11 +21,13 @@
...
             "test_aks_enable_addon_with_azurekeyvaultsecretsprovider",
             "test_aks_enable_addon_with_gitops",
-            "test_aks_create_with_fips"
+            "test_aks_create_with_fips",
+            "test_aks_disable_local_accounts"
         ],
         "unknown": [
             "test_aks_create_and_update_with_managed_aad_enable_azure_rbac",
-            "test_aks_create_with_virtual_node_addon"
+            "test_aks_create_with_virtual_node_addon",
```
Comment (andyliuliming) [inline src/aks-preview/azcli_aks_live_test/ext_matrix_default.json:29]: why exclude this?

**Review by andyliuliming — CHANGES_REQUESTED** [review]
please separate the non-logicial change with the real code chagne.

File: src/aks-preview/azcli_aks_live_test/ext_matrix_default.json (hunk 3 above)
Comment (FumingZhang) [reply src/aks-preview/azcli_aks_live_test/ext_matrix_default.json:31]: tracked in this [wiki]( https://dev.azure.com/msazure/CloudNativeCompute/_wiki/wikis/CloudNativeCompute.wiki/157433/Live-Test-Failures-in-aks-preview-(with-bare-sub)) and this [work item](https://dev.azure.com/msazure/CloudNativeCompute/_workitems/edit/9997354/)

File: src/aks-preview/azcli_aks_live_test/setup_venv.sh (hunk 2 above)
Comment (FumingZhang) [reply src/aks-preview/azcli_aks_live_test/setup_venv.sh]: sure

**Review by andyliuliming — APPROVED** [review]

**Review by Juliehzl — APPROVED** [review]

**Pull request merged**